src/appmixer/kafkadavid/connections.js

```diff
@@ -0,0 +1,177 @@
+        topics, flowId, componentId, groupId, fromBeginning, auth
+    });
+
+    const topicSubscriptions = topics?.AND.map(topic =>
+        topic.topic.startsWith('/') ? RegexParser(topic.topic) : topic.topic
+    );
+
+    auth.connectionTimeout = context.config.connectionTimeout;
+    const client = initClient(auth);
+    debug(context, { type: 'client initialized', auth });;
+    const consumer = client.consumer({ groupId });
+
+    await consumer.connect();
+    OPEN_CONNECTIONS[connectionId] = consumer;
+    debug(context, { type: 'consumer connected', connectionId, OPEN_CONNECTIONS });;
+
+    await consumer.subscribe({
+        topics: topicSubscriptions,
+        fromBeginning: fromBeginning || false
+    });
+
+    debug(context, { type: 'consumer subscribed', topics, fromBeginning });;
+
+    consumer.on(consumer.events.CRASH, async (error) => {
+
+        await context.log('info', 'Kafka consumer CRASH. Removing consumer from open connections list.');
+
+        debug(context, { type: 'consumer CRASH', error });
+
+        delete OPEN_CONNECTIONS[connectionId];
+    });
+
+    await consumer.run({
+        eachMessage: async ({ topic, partition, message, heartbeat, pause }) => {
+
+            debug(context, { type: 'consumer message received', topic, message });;
+
+            const normalizedMessage = normalizeMessageData(message);
+            normalizedMessage.partition = partition;
+            normalizedMessage.pid = process.pid;
+
+            await context.triggerComponent(
+                flowId,
+                componentId,
+                normalizedMessage,
+                { enqueueOnly: true }
+            );
+        }
+    });
+};
+
+const normalizeMessageHeaders = (headers) => {
+
+    const normalizedHeaders = {};
+    if (headers) {
+        Object.entries(headers).forEach(([key, value]) => {
+            normalizedHeaders[key] = Buffer.isBuffer(value) ? value.toString('utf8') : value || '';
+        });
+    }
+    return normalizedHeaders;
+};
+
+const normalizeMessageData = (message) => {
+
+    return {
+        key: message.key ? message.key.toString() : null,
+        value: message.value ? message.value.toString() : null,
+        headers: normalizeMessageHeaders(message.headers)
+    };
+};
+
+const addProducer = async (context, flowId, componentId, auth) => {
+
+    const connectionId = `producer:${flowId}:${componentId}`;
+    await context.service.stateSet(connectionId, {
+        flowId, componentId, auth
+    });
+    auth.connectionTimeout = context.config.connectionTimeout;
+    const client = initClient(auth);
+    const producer = client.producer();
+
+    await producer.connect();
+    OPEN_CONNECTIONS[connectionId] = producer;
+};
+
+const sendMessage = async (flowId, componentId, payload) => {
+
+    const connectionId = `producer:${flowId}:${componentId}`;
+    const producer = OPEN_CONNECTIONS[connectionId];
+    if (!producer) return; // Connection doesn't exist, do nothing
```

Comment:
It should throw an error, that errors get to the component and the message will be retried. Otherwise it is lost.